tough/src/datastore.rs

```diff
@@ -68,6 +89,43 @@ impl Datastore {
             },
         }
     }
+
+    /// Ensures that system time has not stepped backward since it was last sampled. This function
+    /// is protected by a lock guard to ensure thread safety.
+    pub(crate) fn system_time(&self) -> Result<DateTime<Utc>> {
+        // Treat this function as a critical section. This lock is not used for anything else.
+        let _lock = self.time_lock.write().map_err(|e| {
```

Comment:
I always wondered if non-lexical lifetimes meant that this could potentially be dropped prior to the end of the lexical scope.

After reading a few resources, my understanding is that this will be dropped "at the end of its lifetime" which is currently the end of the scope. Possibly it's worth explicitly `drop(_lock)` at the end of the function to make accidental lifetime changes result in a compiler error.